KEP-4247: Update criteria targeting to enable the QueueingHints by default (#4877)
kubernetes/enhancements: keps/sig-scheduling/4247-queueinghint/README.md

- The integration tests are implemented for requeueing scenarios in all plugins.

> - `PreCheck` feature in the scheduling queue is completely removed.

Reviewer: ```suggestion
- `PreCheck` feature in the scheduling queue is disabled when SchedulerQueueingHints is enabled.
```

> - No significant degradation in memory comsumption.

Reviewer: Mention the metric we use to ensure that there are no leaks.

- scheduler_perf covers the performance of most QueueingHintFn for in-tree plugins.

> - scheduler_perf runs with QueueingHint both enabled and disabled for all test cases.

Reviewer: ```suggestion
- scheduler_perf runs with QueueingHint both enabled and disabled for all test cases and throughput when enabled is better or, at least, comparable.
```

- Event handling duration is monitored using scheduler_perf.
- The feature gate is enabled by default.

> - No bug report for a while after enabling it by default.

Reviewer: This should rather be GA criteria, because right now the feature is disabled, so there wouldn't be bug reports.

> 

Reviewer: Have we mentioned any additional criteria that we might consider when going GA?

- Oct 01, 2023: The initial KEP is submitted.
- Dec 13, 2023: The feature gate is changed to be disabled by default.
- Dec 31, 2023: The KEP is updated based on the situation as of v1.30 release cycle. The beta/GA criteria is sorted.

> - Sep 26, 2024: The KEP is updated as QueueingHint is targeting to be enabled by default.

Reviewer: ```suggestion
- Sep 26, 2024: The KEP is updated as QueueingHint is targeting to be enabled by default in the v1.32 release.
```
